projects/storefrontlib/src/lib/pwa/pwa.module.ts

```diff
@@ -8,14 +8,21 @@ import {
 import { Config, ConfigModule } from '@spartacus/core';
 
 import { defaultPWAModuleConfig, PWAModuleConfig } from './pwa.module-config';
-import { AddToHomeScreenBtnComponent } from './add-to-home-screen-btn/add-to-home-screen-btn.component';
+import { AddToHomeScreenBtnComponent } from './components/add-to-home-screen-btn/add-to-home-screen-btn.component';
+import { AddToHomeScreenBannerComponent } from './components/add-to-home-screen-banner/add-to-home-screen-banner.component';
+import { AddToHomeScreenService } from './services/add-to-home-screen.service';
 
 export function pwaConfigurationFactory(
   pwaConfig: PWAModuleConfig
 ): RegistrationOptions {
   return { enabled: (pwaConfig.production && pwaConfig.pwa.enabled) || false };
 }
 
+export function pwaFactory(addToHomeScreenService) {
```

Comment:
Can we remove this pwaFactory and this 
```
{
      provide: APP_INITIALIZER,
      useFactory: pwaFactory,
      deps: [AddToHomeScreenService],
      multi: true
    }
```
Just provide AddToHomeScreenService should be ok, right?

Reply:
It won't work. This service needs to be initialized on app initialization and should be created even when don't have any consumers. `beforeinstallprompt` event can be called in 1 second or after 1 minute. As far as I know this is the only solution.